src/Decred/TransactionBuilder.h

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "Transaction.h"
+#include "../Bitcoin/TransactionPlan.h"
+#include "../Bitcoin/UnspentSelector.h"
+#include "../proto/Bitcoin.pb.h"
+#include "../proto/Decred.pb.h"
+
+#include <algorithm>
+
+namespace TW {
+namespace Decred {
+
+struct TransactionBuilder {
+    /// Plans a transaction by selecting UTXOs and calculating fees.
+    static Bitcoin::TransactionPlan plan(Bitcoin::Proto::SigningInput input) {
```

Comment:
What’s the difference with bitcoin one?

Reply:
TransactionInput and TransactionOutput have extra fields. I tried making this generic but would have made everything harder to understand.

Reply:
I see. because it might be easily overlook `Decred/TransactionBuilder` if we want to update how to construct `TransactionPlan` and I assume we have to update the app side accordingly for Decred

Reply:
UnspentSelector is still the same and if that or TransactionPlan changes you will probably get a compiler error.